packages/styled-components/src/macro/index.js

```diff
@@ -6,12 +6,15 @@ import babelPlugin from 'babel-plugin-styled-components';
 
 function styledComponentsMacro({ references, state, babel: { types: t }, config = {} }) {
   const program = state.file.path;
+  const refNames = Object.keys(references);
 
```

Comment:
@anatoliy8493 pretty sure Object.keys results in another loop internally.

Reply:
reverted that hunk of edits